icefoganalytics/travel-authorization
# Travel Desk: Rank Travel Options: Standardize Travel Desk Components

Relates to:

- https://github.com/icefoganalytics/travel-authorization/issues/230

# Context

The current set of travel desk components are unable to correctly generate flight options.
These components are also so hard to debug that I've decide that standardizing them all will be less work than debugging what's going wrong.

# Implementation

Rebuild a bunch of legacy travel desk components with more traditional patterns.
Move away from deeply nested data, to isolated components.
Fix question creation, and make it possible to add more questions.
Simplify page layout.
Standardize and paginate travel desk table, and support ordering.

NOTE: most changes are structural.

# Screenshots

Travel Desk Page
http://localhost:8080/travel-desk?perPage=5&page=2
![image](https://github.com/user-attachments/assets/e6390d0b-702d-4a99-8730-7a17f563beb0)

Travel Desk Page (with sort)
http://localhost:8080/travel-desk?perPage=5&page=1&sortBy=isBooked.asc&sortBy=isAssignedToCurrentUser.desc&sortBy=travelStartDate.desc
![image](https://github.com/user-attachments/assets/539f3cce-16f4-424e-82c5-6c72341489a5)

Travel Desk Edit Page
http://localhost:8080/travel-desk/38/edit
![image](https://github.com/user-attachments/assets/be9996ce-3ff1-40b1-8207-e27add1d028e)

Travel Desk Edit Page with Add Question Dialog open
http://localhost:8080/travel-desk/38/edit?showTravelDeskQuestionCreate=true
![image](https://github.com/user-attachments/assets/b142e06c-2b85-48ed-9bf5-a9294dd3bf39)

Travel Desk Edit Page with Flight Requests table row expanded to show Flight Options and Flight Segments
http://localhost:8080/travel-desk/1/edit?expandTravelDeskFlightRequest=124
![image](https://github.com/user-attachments/assets/11170abc-19b4-4291-944a-c5ea492b0a57)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Fill out a travel request until after you have submitted it to the travel desk.
5. Log in as a travel desk user, and go to the "Travel Desk" page from the top drop down nav.
6. Using testing instructions from https://github.com/icefoganalytics/travel-authorization/pull/237 check that travel desk page mostly works.
